src/csl/mod.rs

```diff
@@ -853,25 +853,39 @@ fn collapse_items<'a, T: EntryLike>(cite: &mut SpeculativeCiteRender<'a, '_, T>)
     match style.citation.collapse {
         Some(Collapse::CitationNumber) => {
             // Option with the start and end of the range.
+            let after_collapse_delim =
```

Comment:
Might be worth moving this line upwards, as it is shared by the two non-empty match branches.